Platform/CoffeelakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -1036,42 +1089,9 @@ BoardInit (
     EnableLegacyRegions ();
     GpioInit ();
     SpiConstructor ();
+    UpdatePayloadId ();
 
-    PayloadId = GetPayloadId ();
-    GenericCfgData = (GEN_CFG_DATA *)FindConfigDataByTag (CDATA_GEN_TAG);
-    if (GenericCfgData != NULL) {
-      if (GenericCfgData->PayloadId == AUTO_PAYLOAD_ID_SIGNATURE) {
-        PayloadId = 0;
-      } else {
-        PayloadId = GenericCfgData->PayloadId;
-      }
-    }
-
-    //
-    // Switch payloads based on configured GPIO pin
-    //
-    SiliconCfgData = (SILICON_CFG_DATA *)FindConfigDataByTag (CDATA_SILICON_TAG);
-    if ((SiliconCfgData != NULL) && (SiliconCfgData->PayloadSelGpio.Enable != 0)){
-      if (IsPchLp() == TRUE) {
-        PayloadSelGpioPad = GPIO_CFG_PIN_TO_PAD(SiliconCfgData->PayloadSelGpio) | (GPIO_CNL_LP_CHIPSET_ID << 24);
-      } else {
-        PayloadSelGpioPad = GPIO_CFG_PIN_TO_PAD(SiliconCfgData->PayloadSelGpio) | (GPIO_CNL_H_CHIPSET_ID << 24);
-      }
-      Status = GpioGetInputValue (PayloadSelGpioPad, &PayloadSelGpioData);
-      if (!EFI_ERROR (Status)) {
-        if (PayloadSelGpioData == 0) {
-          PayloadId = 0;
-        } else {
-          if ((GenericCfgData != NULL) && (GenericCfgData->PayloadId == AUTO_PAYLOAD_ID_SIGNATURE)) {
-            PayloadId = UEFI_PAYLOAD_ID_SIGNATURE;
-          }
-        }
-        DEBUG ((DEBUG_INFO, "Set PayloadId to 0x%08X based on GPIO config\n", PayloadId));
-      }
-    }
-    mSiliconCfgData = SiliconCfgData;
-
-    SetPayloadId (PayloadId);
+    mSiliconCfgData = (SILICON_CFG_DATA *)FindConfigDataByTag (CDATA_SILICON_TAG);
```

Comment:
mSiliconCfgData is required? Even if current stage is Stage2, but better to avoid global variable.  UpdatePayloadId () also does not use mSiliconCfgData.

Reply:
agree.

Reply:
Done, it is used by platformupdateacpitable, not sure why it is used that way, so left untouched. but removed now